src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/HMPPSAuthService.kt

```diff
@@ -98,15 +100,15 @@ class HMPPSAuthService(
           .retrieve()
           .bodyToMono(UserDetailResponse::class.java)
           .withRetryPolicy()
-          .map { it.name.substringBefore(' ') },
+          .map { Pair(it.name.substringBefore(' '), it.name.substringAfterLast(' ')) },
```

Comment:
_nit_
Would using `it.name.split` with a limit make this simpler?

Reply:
possibly - names could be any number of words e.g. tom david lantos myers, this code just plucks the first and last words out. the alternative would be `it.name.split(' ').first` ... `it.name.split(' ').last` or something - whcih i don't think is any neater. do you?

Reply:
👍 yeah, this is fine, thanks